tasklist/els-schema/src/main/java/io/camunda/tasklist/es/IndexSchemaValidatorElasticSearch.java

```diff
@@ -164,4 +224,86 @@ public boolean validateNumberOfReplicas(final List<String> indexes) {
     }
     return true;
   }
+
+
+  private IndexMappingDifference getDifference(
+      final IndexDescriptor indexDescriptor, final Map<String, IndexMapping> indexMappingsGroup) {
+    return getIndexMappingDifference(indexDescriptor, indexMappingsGroup);
+  }
+
+  private void validateDifferenceAndCollectNewFields(
+      final IndexDescriptor indexDescriptor,
+      final IndexMappingDifference difference,
+      final Map<IndexDescriptor, Set<IndexMappingProperty>> newFields) {
+    if (difference != null && !difference.isEqual()) {
```

Comment:
According to the code, `!difference.isEqual()` can never be false in case `difference != null`

Reply:
You are correct, This works, but the verification is redundant.
I will provide an update to make this more readable and less redundant on this commit. https://github.com/camunda/zeebe/pull/18618/commits/907211eee77d07914338aab708dea838084787d0